src/app/pages/storage/volumes/volume-import-wizard/volume-import-wizard.component.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { createComponentFactory, mockProvider, Spectator } from '@ngneat/spectator/jest';
+import { fakeSuccessfulJob } from 'app/core/testing/utils/fake-job.utils';
+import { mockJob, mockWebsocket } from 'app/core/testing/utils/mock-websocket.utils';
+import { PoolStatus } from 'app/enums/pool-status.enum';
+import { Option } from 'app/interfaces/option.interface';
+import { PoolFindResult } from 'app/interfaces/pool-import.interface';
+import { EntityJobComponent } from 'app/modules/entity/entity-job/entity-job.component';
+import { IxFormsModule } from 'app/modules/ix-forms/ix-forms.module';
+import { IxFormHarness } from 'app/modules/ix-forms/testing/ix-form.harness';
+import { WebSocketService, DialogService } from 'app/services';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+import { VolumeImportWizardComponent } from './volume-import-wizard.component';
+
+describe('VolumeImportWizardComponent', () => {
+  let spectator: Spectator<VolumeImportWizardComponent>;
+  let loader: HarnessLoader;
+  let ws: WebSocketService;
+  const mockDialogRef = {
+    componentInstance: {
+      setDescription: jest.fn(),
+      setCall: jest.fn(),
+      submit: jest.fn(),
+      success: new EventEmitter(),
+      failure: new EventEmitter(),
+    } as unknown as EntityJobComponent,
+  } as MatDialogRef<EntityJobComponent, any>;
+
+  const createComponent = createComponentFactory({
+    component: VolumeImportWizardComponent,
+    imports: [
+      IxFormsModule,
+      ReactiveFormsModule,
+    ],
+    providers: [
+      mockWebsocket([
+        mockJob('pool.import_pool', fakeSuccessfulJob()),
+        mockJob('pool.import_find',
+          fakeSuccessfulJob([{
+            name: 'pool_name_1',
+            guid: 'pool_guid_1',
+            hostname: 'pool_hostname_1',
+            status: PoolStatus.Online,
+          }, {
+            name: 'pool_name_2',
+            guid: 'pool_guid_2',
+            hostname: 'pool_hostname_2',
+            status: PoolStatus.Online,
+          }, {
+            name: 'pool_name_3',
+            guid: 'pool_guid_3',
+            hostname: 'pool_hostname_3',
+            status: PoolStatus.Online,
+          }] as PoolFindResult[])),
+      ]),
+      mockProvider(IxSlideInService),
+      mockProvider(DialogService),
+      mockProvider(MatDialog, {
+        open: () => mockDialogRef,
+      } as unknown as MatDialog),
+    ],
+  });
+
+  beforeEach(() => {
+    spectator = createComponent();
+    loader = TestbedHarnessEnvironment.loader(spectator.fixture);
+    ws = spectator.inject(WebSocketService);
+  });
+
+  it('loads and shows the current list of pools to import when form is opened', (done) => {
+    spectator.component.pool.options.subscribe((options: Option[]) => {
+      expect(ws.job).toHaveBeenCalledWith('pool.import_find');
+      expect(options).toEqual([
```

Comment:
This is basically checking for a static variable twice since our mockup is returning this result. Instead maybe we should test the template element and see if that got the options.